src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String OPERATION_RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supply = 0;
+        int buy = 0;
+        String[] dataFromFile = readDataFromFile(fromFileName);
+        for (int i = 0; i < dataFromFile.length; i += 2) {
+            switch (dataFromFile[i]) {
+                case OPERATION_SUPPLY:
+                    supply += Integer.parseInt(dataFromFile[i + 1]);
+                    break;
+                case OPERATION_BUY:
+                    buy += Integer.parseInt(dataFromFile[i + 1]);
+                    break;
+                default:
+                    throw new IllegalStateException("Unexpected value: " + dataFromFile[i]);
+            }
+        }
+        String statisticData = OPERATION_SUPPLY + "," + supply + System.lineSeparator()
+                + OPERATION_BUY + "," + buy + System.lineSeparator()
+                + OPERATION_RESULT + "," + (supply - buy);
+        writeDataToFile(toFileName, statisticData);
```

Comment:
You correctly made separate methods for reading and writing. Let's go ahead and extract a separate method for report generating.
```suggestion
        String[] dataFromFile = readDataFromFile(fromFileName);
        String report = generateReport(dataFromFile);
        writeDataToFile(toFileName, report);
```